src/pymor/operators/numpy.py

```diff
@@ -396,3 +397,53 @@ def _format_repr(self, max_width, verbosity):
         else:
             matrix_repr = f'<{self.range.dim}x{self.source.dim} dense>'
         return super()._format_repr(max_width, verbosity, override={'matrix': matrix_repr})
+
+
+class NumpyHankelOperator(NumpyGenericOperator):
+    def __init__(self, markov_parameters, source_id=None, range_id=None, solver_options=None, name=None):
+        markov_parameters = np.squeeze(markov_parameters)
+        assert markov_parameters.ndim == 1
+        try:
+            markov_parameters.setflags(write=False)  # make numpy arrays read-only
+        except AttributeError:
+            pass
+
+        self.__auto_init(locals())
+        self.source = NumpyVectorSpace(ceil(len(markov_parameters) / 2), source_id)
+        self.range = NumpyVectorSpace(ceil(len(markov_parameters) / 2), range_id)
+        self.linear = True
+        self._circulant = self._calc_circulant()
+
+    def apply(self, U, mu=None):
+        assert U in self.source
+        U = U.to_numpy().T
+        k = U.shape[1]
+        x = np.concatenate([np.flip(U, axis=0), np.zeros([self.source.dim, k])])
+        y = np.real(ifft(fft(x, axis=0,) * self._circulant, axis=0))
+        return self.range.make_array(y[: self.source.dim].T)
+
+    def apply_adjoint(self, V, mu=None):
+        assert V in self.range
+        return self.H.apply(V, mu=mu)
+
+    def _calc_circulant(self):
+        pad = len(self.markov_parameters) % 2
+        c = np.roll(
+            np.concatenate([np.zeros(pad), self.markov_parameters], axis=-1,),
+            self.source.dim + 1 - pad,
```

Comment:
Maybe this simplifies it a bit:

```suggestion
            len(self.markov_parameters) // 2 + 1,
```